Source/Core/DolphinTool/VerifyCommand.cpp

```diff
@@ -80,34 +76,36 @@ int VerifyCommand::Main(const std::vector<std::string>& args)
   }
 
   // Open the volume
-  std::shared_ptr<DiscIO::VolumeDisc> volume = DiscIO::CreateDisc(input_file_path);
+  const std::unique_ptr<DiscIO::VolumeDisc> volume = DiscIO::CreateDisc(input_file_path);
   if (!volume)
   {
     std::cerr << "Error: Unable to open disc image" << std::endl;
     return 1;
   }
 
   // Verify the volume
-  const std::optional<DiscIO::VolumeVerifier::Result> result =
-      VerifyVolume(volume, hashes_to_calculate);
-  if (!result)
+  DiscIO::VolumeVerifier verifier(*volume, false, hashes_to_calculate);
+  verifier.Start();
+  while (verifier.GetBytesProcessed() != verifier.GetTotalBytes())
```

Comment:
Why did you pull this out of a function? This feels worse to me than before.

Reply:
Do you want to avoid the copy of the `DiscIO::VolumeVerifier::Result`? In that case you forgot a `&`.

Reply:
I pulled it out of a function because it was wholly unnecessary for it to be outlined in the first place.  For starters, It is only used once.  It introduced an unnecessary failure state in the event that the DiscIO::VolumeDisc is a nullptr, which it never can be.  Also, implementing the function inline took almost the same number of lines as it already took to call and check the result of VerifyCommand::VerifyVolume.

Reply:
And also this avoids copying result, which I didn't know was returned by const reference until you just mentioned it.

Reply:
A function can be written simply for clarity even if it is only used once, or under the assumption that the functionality might be useful in another use case as well even if it doesn't exist at the moment.

Whatever, I think in this case it's not much of a difference either way, but your arguments feel a bit questionable to me.